routes/auth.js

```diff
@@ -0,0 +1,100 @@
+var express = require('express');
+var router = express.Router();
+var utils = require('../utils.js');
+var ldapUtils = require('../ldap.js');
+
+module.exports = function(connection) {
+
+    /* Get all auth rules */
+    router.get('/auth', function (req, res, next) {
+
+        connection.query('select * from auth', function (error, rows, field) {
+
+            if(!!error){
+                console.log(error);
+            }
+            else {
+                res.setHeader('Content-Type', 'application/json');
+                res.json(rows);
+            }
+        })
+    });
+    
+    /* Create new auth rule */
+    router.post('/auth', utils.isAuthenticated, function (req, res, next) {
+        var atlas_id=req.body.atlas_id;
+
```

Comment:
If the principal is blank/empty, return a `400 Bad Request` and exit before querying ldap.